### PR TITLE
Add Schedule Feature (Partial)

### LINKED ISSUE #153: Add Skeleton for Schedule Commands
- Added Event (an abstract superclass of Appointment and Schedule). Purpose of this superclass to allow the new proposed Timetable Ui to show both Appointment and Schedule.
- Slight modification to Appointment to inherit from Event
- Added Schedule and its respective variable classes
- Added EventList, ScheduleList and ScheduleTracker
- Added AddScheduleCommand and DeleteScheduleCommand and its respective Parser.

### PR DESCRIPTION
**Pull request by glatiuden**
Part 1 of #133

Changelog
- Added `Event` (an abstract superclass of `Appointment` and `Schedule`). Purpose of this superclass to allow the new proposed Timetable Ui to show both `Appointment` and `Schedule`.
- Slight modification to `Appointment` to inherit from `Event`
- Added `Schedule` and its respective variable classes
- Added `EventList`, `ScheduleList` and `ScheduleTracker` 
- Added `AddScheduleCommand` and `DeleteScheduleCommand` and its respective Parser.

Upon merging this PR, the next PR will include (Completed):
- `ListScheduleCommand` and `EditScheduleCommand` and its respective parsers
- Timetable Ui

In the upcoming PR:
- Modify `ModelManager` constructor to take in `ScheduleTracker` (not implemented in this PR to avoid huge PR)
- JUnit test cases for the new Java classes
- `Reminder` and its respective commands

### CODE REVIEW

**Comment by jared98lyj** [comment]
LGTM

**Review by jared98lyj — DISMISSED** [review]

File: src/main/java/seedu/address/model/Model.java [hunk 1]
```diff
@@ -8,11 +8,12 @@
 import seedu.address.model.appointment.Appointment;
 import seedu.address.model.appointment.AppointmentDateTime;
 import seedu.address.model.person.Person;
+import seedu.address.model.schedule.ScheduleModel;
 
 /**
  * The API of the Model component.
  */
-public interface Model {
+public interface Model extends ScheduleModel {
```
Comment (kingsleykuan) [inline src/main/java/seedu/address/model/Model.java]: I am slightly concerned about having Model extend ScheduleModel in terms of design, but I understand why it's done this way for abstraction and to maintain compatibility. If there is no better solution then it LGTM. Perhaps in a future PR we can abstract the different model interfaces out (for tutor, appointment, etc) and then have Model extend all of them for consistency.

**Review by kingsleykuan — DISMISSED** [review]
LGTM but it seems there are some merge conflicts. Also see my comment below (no change necessarily required).

File: src/main/java/seedu/address/model/Model.java (hunk 1 above)
Comment (glatiuden) [reply src/main/java/seedu/address/model/Model.java]: Noted, then I'll shift it back to Model for meantime. If we decided to adopt this approach then I'll reinstate back in this file

**Comment by glatiuden** [comment]
Fixed merge conflicts and shifting back the interface methods to `Model.java`

**Comment by glatiuden** [comment]
Fixed merge conflicts and shifting of Schedule's interface methods to `Model.java`.

**Review by kingsleykuan — APPROVED** [review]
LGTM

**Comment by jared98lyj** [comment]
LGTM

**Review by jared98lyj — APPROVED** [review]
LGTM

**Pull request merged**